scipy/spatial/qhull.pyx

```diff
@@ -2406,9 +2523,183 @@ class Voronoi(_QhullUser):
+        Volume of the dual convex hull
+
+    Raises
+    ------
+    QhullError
+        Raised when Qhull encounters an error condition, such as
+        geometrical degeneracy when options to resolve are not enabled.
+    ValueError
+        Raised if an incompatible array is given as input.
+
+    Notes
+    -----
+    The intersections are computed using the 
+    `Qhull library <http://www.qhull.org/>`__.
+
+    Examples
+    --------
+
+    Halfspace intersection of planes forming some polygon
+
+    >>> from scipy.spatial import HalfspaceIntersection
+    >>> import numpy as np
+    >>> halfspaces = np.array([[-1, 0., 0.],
+    ...                        [0., -1., 0.],
+    ...                        [2., 1., -4.],
+    ...                        [-0.5, 1., -2.]])
+    >>> feasible_point = np.array([0.5, 0.5])
+    >>> hs = HalfspaceIntersection(halfspaces, feasible_point)
+
+    Plot halsfpaces as filled regions and intersection points:
+
+    >>> import matplotlib.pyplot as plt
+    >>> xlim, ylim = (-1, 3), (-2, 6)
+    >>> x = np.linspace(-1, 3, 100)
+    >>> symbols = ['-', '+', 'x', '*']
+    >>> signs = [0, 0, -1, -1]
+    >>> fmt = {"color": None, "edgecolor": "b", "alpha": 0.5}
+    >>> for h, sym, sign in zip(halfspaces, symbols, signs):
+    ...     hlist = h.tolist()
+    ...     fmt["hatch"] = sym
+    ...     if h[1]== 0:
+    ...         plt.axvline(-h[2]/h[0], label='{}x+{}y+{}=0'.format(*hlist))
+    ...         xi = np.linspace(xlim[sign], -h[2]/h[0], 100)
+    ...         plt.fill_between(xi, ylim[0], ylim[1], **fmt)
+    ...     else:
+    ...         plt.plot(x, (-h[2]-h[0]*x)/h[1], label='{}x+{}y+{}=0'.format(*hlist))
+    ...         plt.fill_between(x, (-h[2]-h[0]*x)/h[1], **fmt)
+    >>> x, y = zip(*hs.intersections)
+    >>> plt.plot(x, y, 'o', markersize=8)
+    >>> plt.legend()
+    >>> plt.show()
+
+    References
+    ----------
+    .. [Qhull] http://www.qhull.org/
+
+    """
+
+    def __init__(self, halfspaces, feasible_point,
+                    incremental=False, qhull_options=None):
+        if np.ma.isMaskedArray(halfspaces):
+            raise ValueError('Input halfspaces cannot be a masked array')
+        if np.ma.isMaskedArray(feasible_point):
+            raise ValueError('Input feasible point cannot be a masked array')
+        if feasible_point.ndim != 1 or feasible_point.shape[0] != halfspaces.shape[1]-1:
+            raise ValueError('Feasible point must be a (ndim-1,) array')
+        halfspaces = np.ascontiguousarray(halfspaces, dtype=np.double)
+        self.feasible_point = np.ascontiguousarray(feasible_point, dtype=np.double)
+
+        if qhull_options is None:
+            qhull_options = b""
+            if halfspaces.shape[1] >= 6:
+                qhull_options += b"Qx"
+        else:
+            qhull_options = asbytes(qhull_options)
+
+        # Run qhull
+        mode_option = "H{}".format(','.join([str(self.feasible_point.item(i)) for i in range(halfspaces.shape[1] -1)]))
```

Comment:
Well if we have to patch qhull, wouldn't it be easier to simply add another double* argument to qh_new_qhull that defaults to NULL and if not null, is used to fill qh->feasible_point?